### DISCUSSION TITLE
Статуэтка Дарона

### DISCUSSION THREAD

**Discussion by dosinabox** [dosinabox/g2nr_unofficial_update#504] [Ideas]
Нужно сделать возможность отдать статуэтку Дарону, если уже вступил в любую гильдию (включая послушника).

**Comment by Kor-Angar** [comment]
Статуэтку отдаваит Дарону или в Монастырь?

**Comment by dosinabox** [comment]
Послушника отправлять в монастырь. У всех остальных пусть Дарон забирает и закрывает квест.
Послушник - мальчик на побегушках. Мага и паладина гонять несолидно. Всем остальным в монастырь нельзя.

**Comment by Kor-Angar** [comment]
Ты хочешь прям квест сделать с заданием или просто возможность сдать статуэтку?

**Comment by dosinabox** [comment]
- Дарон заберет статуэтку у ГГ-мага (гонять несолидно) или в третьей главе (потому что пропадает Педро);
- При вступлении в любую гильдию невыполненный квест "Простой путь в монастырь" превратится в новый отдельный квест "Пропавшая статуэтка";
- Дарон расскажет о статуэтке ГГ любой гильдии, если ГГ не узнал об этом у Ватраса, будучи нейтралом (диалог "У тебя такой расстроенный вид...");
- Педро не скажет ГГ-послушнику или магу, что его путь в монастырь закрыт;
- ГГ-послушник может принести Гораксу статуэтку когда угодно и вступить в монастырь бесплатно, если ещё не заплатил 1000 золотых;
- ГГ-послушник (заплативший 1000 золотых) или ГГ-маг отдает статуэтку Педро (у Горакса нет подходящих реплик);
- Если Педро пропал, то ГГ отдает статуэтку Гораксу;
- Если отдать статуэтку Педро, то она останется в его инвентаре на Ирдорате (?) :)

**Comment by dosinabox** [comment]
Вроде как готово, нужно проверить.

**Comment by Kor-Angar** [comment]
а почему ты не хочешь просто что бы ГГ отдал эту статуэтку Дарону в городе?

**Comment by dosinabox** [comment]
ГГ отдаст в третьей главе. ГГ-маг отдаст всегда. Для остальных не хотелось бы ограничивать контент (пропадет диалог с Педро).

**Comment by Kor-Angar** [comment]
> Как в этом прохождении поступил со статуэткой Дарона?

пытался с Дароном/Педро/Гораксом про неё поговорить, диалога не было.

**Comment by dosinabox** [comment]
Нужны подробности.

**Comment by dosinabox** [comment]
И, желательно, сохраненка.

**Comment by Kor-Angar** [comment]
проверял. у Мага всё Ок
2 часа 14 минут
https://youtu.be/1MkUKPijVhY?t=8040

**Comment by Kor-Angar** [comment]
Играл Наёмником. Принёс статуэтку Педро. Денег Педро не взял. Сказал какую то фразу про то что в монастырь мне нельзя. Не понял почему Дарон у Наёмника не принял статуэтку.

**Comment by dosinabox** [comment]
На самом деле, под эту ситуацию даже написан отдельный диалог:
```
instance DIA_Addon_Pedro_Statuette_Abgeben(C_Info)
{
	npc = NOV_600_Pedro;
	nr = 2;
	condition = DIA_Addon_Pedro_Statuette_Abgeben_Condition;
	information = DIA_Addon_Pedro_Statuette_Abgeben_Info;
	permanent = FALSE;
	description = "Я могу отдать статуэтку тебе?";
};


func int DIA_Addon_Pedro_Statuette_Abgeben_Condition()
{
	if(Npc_HasItems(other,ItMi_LostInnosStatue_Daron) && Npc_KnowsInfo(other,DIA_Addon_Pedro_Statuette) && (LostInnosStatueInMonastery == FALSE))
	{
		if((hero.guild != GIL_NONE) && (hero.guild != GIL_NOV) && (hero.guild != GIL_KDF))
		{
			return TRUE;
		};
	};
};
```
Странно, если не сработало. Есть запись момента?

**Comment by dosinabox** [comment]
И даже Дарон должен её принимать.
```
func int DIA_Addon_Daron_FoundStatue_Condition()
{
	if(Npc_HasItems(other,ItMi_LostInnosStatue_Daron) && (MIS_Addon_Daron_GetStatue == LOG_Running) && (LostInnosStatueInMonastery == FALSE))
	{
		return TRUE;
	};
};

func void DIA_Addon_Daron_FoundStatue_Info()
{
	AI_Output(other,self,"DIA_Addon_Daron_FoundStatue_15_00");	//Я нашел статуэтку.
	if((other.guild == GIL_KDF) || (MIS_OLDWORLD == LOG_SUCCESS))
	{
		B_GiveInvItems(other,self,ItMi_LostInnosStatue_Daron,1);
		Npc_RemoveInvItem(self,ItMi_LostInnosStatue_Daron);
		B_DaronGivesReward();
	}
	else
	{
		AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_01");	//Хвала Инносу!
		AI_Output(other,self,"DIA_Addon_Daron_FoundStatue_15_02");	//Что ты будешь с ней делать?
		AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_03");	//Я? Ничего. До сих пор она приносила мне лишь несчастья.
		AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_04");	//ТЫ отнесешь ее в монастырь, сынок.
		if(other.guild == GIL_NONE)
		{
			AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_05");	//Конечно, тебя не пропустят внутрь, если ты не согласишься посвятить оставшуюся жизнь служению монастырю.
			AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_06");	//Но я уверен, что ты не откажешься. Ведь ты можешь оказать услугу МНЕ, верно?
		};
		AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_07");	//Ступай, сынок! Да пребудет с тобой Иннос.
	};
};
```

**Comment by Kor-Angar** [comment]
https://youtu.be/pLK4fytboH8

1 час 47 минут

**Comment by Kor-Angar** [comment]
> И даже Дарон должен её принимать

Я там про Наемника писал, а в коде он у Мага принимает. Или я неправильно его прочитал?

**Comment by dosinabox** [comment]
А запись диалога с Педро есть?

**Comment by Kor-Angar** [comment]
Нету. Добиручь до клипа, сделаю полностью.

А почему ты просто не хочешь что бы герой сдавал статуэтку и получал от Дарона благословение(бонус к жизни) и/или золото?

**Comment by Kor-Angar** [comment]
Нет, но можно сделать8:16, 15 октября 2019 г., dosinabox <notifications@github.com>:А запись диалога с Педро есть?

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
-- Отправлено из мобильного приложения Яндекс.Почты

**Comment by dosinabox** [comment]
> Я там про Наемника писал, а в коде он у Мага принимает. Или я неправильно его прочитал?

Сразу забирает у мага.
Сразу забирает у любого, кто принес Хагену письмо Гаронда.
Остальных гоняет в монастырь.

> А почему ты просто не хочешь что бы герой сдавал статуэтку и получал от Дарона благословение(бонус к жизни) и/или золото?

Как бы такая задумка и была. Но условие с письмом получилось странное (?) (мы это уже обсуждали? я не помню, откуда взялось это условие). Наверное, гонять в монастырь нужно только нейтрала и послушника, а у остальных забирать сразу.

**Comment by Kor-Angar** [comment]
> Но условие с письмом получилось странное (?) (мы это уже обсуждали? я не помню, откуда взялось это условие). 

нет. не обсуждали. условие странное, т.к. получается что 

- у Мента/Наёма который уже идёт в монастырь, Дарон статуэтку забирает
- у Мента/Наёма который не идёт в монастырь, Дарон статуэтку не забирает

видимо просто потому что Педро исчезает когда ГГ получает письмо

> Наверное, гонять в монастырь нужно только нейтрала и послушника, а у остальных забирать сразу.

Нейтрала в рамках миссии по быстрому вступлению? или нейтрала в любом случае?

**Comment by dosinabox** [comment]
> видимо просто потому что Педро исчезает когда ГГ получает письмо

Да, точно

**Comment by dosinabox** [comment]
> Нейтрала в рамках миссии по быстрому вступлению? или нейтрала в любом случае?

Нейтрал выходит на этот квест по наводке от Ватраса, как в оригинале.
В обновлении добавляются дополнительные возможности взять квест:
- послушник спрашивает у Дарона, есть ли для него задания
- все остальные просто спрашивают у Дарона как у него дела, а он в ответ начинает плакаться, что потерял статуэтку

**Comment by Kor-Angar** [comment]
теперь ясно. награда от Дарона какая будет?

**Comment by dosinabox** [comment]
> теперь ясно. награда от Дарона какая будет?

Всегда одинаковая - 150 золотых для всех.

**Comment by dosinabox** [comment]
Ты для проверки просто сходи в монастырь и отдай статуэтку Педро. Если он уже пропал, то можно будет отдать Гораксу. Можешь проверить оба случая.

**Comment by Kor-Angar** [comment]
Я сдавал. Он её взял. Денег не дал. 14:18, 15 октября 2019 г., dosinabox <notifications@github.com>:Ты для проверки просто сходи в монастырь и отдай статуэтку Педро. Если он уже пропал, то можно будет отдать Гораксу. Можешь проверить оба случая.

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
-- Отправлено из мобильного приложения Яндекс.Почты

**Comment by Kor-Angar** [comment]
Или мне так показалось. Перепроверю14:18, 15 октября 2019 г., dosinabox <notifications@github.com>:Ты для проверки просто сходи в монастырь и отдай статуэтку Педро. Если он уже пропал, то можно будет отдать Гораксу. Можешь проверить оба случая.

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
-- Отправлено из мобильного приложения Яндекс.Почты

**Comment by dosinabox** [comment]
Ты там, кажется, загружал старую сохраненку, потому что давно не сохранялся. Может, перепутал при повторном прохождении.

**Comment by dosinabox** [comment]
> Я сдавал. Он её взял. Денег не дал.

Стоп. Если Педро забрал у тебя статуэтку, то всё в порядке. Теперь нужно вернуться к Дарону, который скажет тебе спасибо и даст 150 золотых.

**Comment by Kor-Angar** [comment]
мент. 2 глава. с Дароном поговорил о статуэтке. он отправил меня в монастырь Инноса отнести её. так задумано? я опять повторяюсь, но кажется я запутался :-)

**Comment by dosinabox** [comment]
правильно

**Comment by Kor-Angar** [comment]
у Горакса диалог появляется про статуэтку появляется до того как Дарон поручил её отнести в монастырь. ОнД

**Comment by Kor-Angar** [comment]
а ты не думала что бы вот этих троих гоблинов в левой части пещеры, вставлять после получения квеста от Дарона?

**Comment by dosinabox** [comment]
> у Горакса диалог появляется про статуэтку появляется до того как Дарон поручил её отнести в монастырь. ОнД

```cpp
func int DIA_Addon_Gorax_DaronsStatue_Condition()
{
	if((other.guild == GIL_NOV) && (DIA_Gorax_GOLD_perm == FALSE) && (Pedro_NOV_Aufnahme_LostInnosStatue_Daron == TRUE))
	{
		return TRUE;
	}
	else if((MIS_OLDWORLD == LOG_SUCCESS) && (MIS_Addon_Daron_GetStatue == LOG_Running) && (LostInnosStatueInMonastery == FALSE))
	{
		return TRUE;
	};
};
```

Тут второе условие. Ошибок не вижу, `MIS_Addon_Daron_GetStatue` становится `LOG_Running` только в диалоге с Дароном "Я услышал достаточно. Я найду статуэтку." (`DIA_Addon_Daron_GuildHelp_auftrag`).

> а ты не думала что бы вот этих троих гоблинов в левой части пещеры, вставлять после получения квеста от Дарона?

Нет. Наверное, это лишнее.

**Comment by Kor-Angar** [comment]
Так поручение отнести статуэтку в монастырь даётся после того как герой приносит статуэтку Дарону.

**Comment by dosinabox** [comment]
> Так поручение отнести статуэтку в монастырь даётся после того как герой приносит статуэтку Дарону.

Нет, есть ещё "У тебя такой расстроенный вид".

**Comment by dosinabox** [comment]
А для послушника - "У тебя есть какое-нибудь задание для меня?"

**Comment by Kor-Angar** [comment]
мы про разное. 
Вот Дарон даёт квест. Герой находит статуэтку, приходит  к Дарону и тот его отправляет в Монастырь.

у тебя статуэтку можно отдавать Гораксу до того как Дарон отправить героя отнести статуэтку в Монастырь.
ты делаешь проверку по MIS_Addon_Daron_GetStatue == LOG_Running если добавить то такой ситуации не будет Npc_KnowsInfo(other,DIA_Addon_Daron_FoundStatue)

**

сейчас у меня ОнД в 4 главе весит этот квест, выполнить я его не могу, а у Горакса весит данный диалог.

**Comment by Kor-Angar** [comment]
> reopened this yesterday

опять правки? :-)

**Comment by dosinabox** [comment]
> у тебя статуэтку можно отдавать Гораксу до того как Дарон отправить героя отнести статуэтку в Монастырь.
> ты делаешь проверку по MIS_Addon_Daron_GetStatue == LOG_Running если добавить то такой ситуации не будет Npc_KnowsInfo(other,DIA_Addon_Daron_FoundStatue)

нужно не забыть и изучить это подробнее

**Comment by Kor-Angar** [comment]
вот диалог Дарона

```D
instance DIA_Addon_Daron_FoundStatue(C_Info)
{
	npc = KDF_511_Daron;
	nr = 5;
	condition = DIA_Addon_Daron_FoundStatue_Condition;
	information = DIA_Addon_Daron_FoundStatue_Info;
	description = "Я нашел статуэтку.";
};


func int DIA_Addon_Daron_FoundStatue_Condition()
{
	if(Npc_HasItems(other,ItMi_LostInnosStatue_Daron) && (MIS_Addon_Daron_GetStatue == LOG_Running) && (LostInnosStatueInMonastery == FALSE))
	{
		return TRUE;
	};
};

func void DIA_Addon_Daron_FoundStatue_Info()
{
	AI_Output(other,self,"DIA_Addon_Daron_FoundStatue_15_00");	//Я нашел статуэтку.
	if((other.guild == GIL_KDF) || (MIS_OLDWORLD == LOG_SUCCESS))
	{
		B_GiveInvItems(other,self,ItMi_LostInnosStatue_Daron,1);
		Npc_RemoveInvItem(self,ItMi_LostInnosStatue_Daron);
		B_DaronGivesReward();
	}
	else
	{
		AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_01");	//Хвала Инносу!
		AI_Output(other,self,"DIA_Addon_Daron_FoundStatue_15_02");	//Что ты будешь с ней делать?
		AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_03");	//Я? Ничего. До сих пор она приносила мне лишь несчастья.
		AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_04");	//ТЫ отнесешь ее в монастырь, сынок.
		if(other.guild == GIL_NONE)
		{
			AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_05");	//Конечно, тебя не пропустят внутрь, если ты не согласишься посвятить оставшуюся жизнь служению монастырю.
			AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_06");	//Но я уверен, что ты не откажешься. Ведь ты можешь оказать услугу МНЕ, верно?
		};
		AI_Output(self,other,"DIA_Addon_Daron_FoundStatue_10_07");	//Ступай, сынок! Да пребудет с тобой Иннос.
	};
};
```
а вот условие на диалог с Гораксом

```D
func int DIA_Addon_Gorax_DaronsStatue_Condition()
{
	if((other.guild == GIL_NOV) && (DIA_Gorax_GOLD_perm == FALSE) && (Pedro_NOV_Aufnahme_LostInnosStatue_Daron == TRUE))
	{
		return TRUE;
	}
	else if((MIS_OLDWORLD == LOG_SUCCESS) && (MIS_Addon_Daron_GetStatue == LOG_Running) && (LostInnosStatueInMonastery == FALSE))
	{
		return TRUE;
	};
};
```
смотрим внимательно и видим, что при условие _(MIS_OLDWORLD == LOG_SUCCESS)_ статуэтка просто сдаётся Дарону в городе. 
```
if((other.guild == GIL_KDF) || (MIS_OLDWORLD == LOG_SUCCESS))
```
т.е. оно уже вначале запараллелено и решает некую виртуальную задачу. Проще у Горакса делать проверку на смерть Дарона, тогда это будет решать реальную задачу.

весь твой квест новый получается за пределами сюжетной линии:-) и это не голословно. я три прохождения пытался его пройти и каждый раз у меня был взрыв мозга. и я скажу почему: ты не играешь в свои правки. это теория из скриптов создаёт в игре ещё те парадоксы.

**Comment by Kor-Angar** [comment]
### монастырь огня
Маг сдаёт статуэтку Дарону
для Послушника отдельные условия на диалог с Гораксом
### не монастырь огня
в 3 главе  сдают статуэтку  Дарону. До 3 главы не могут поговорить с Гораксом.

***
при каких игровых условиях ОнД/Наём/Пал/Мент/Нейтрал должны говорить с Гораксом? :-)

**Comment by Kor-Angar** [comment]
<Цензура>

**Comment by Kor-Angar** [comment]
что опять есть диалог про статуэтку без статуэтки
10 минута стрим https://www.youtube.com/watch?v=Jwd0vfWoKq0

**Comment by dosinabox** [comment]
Пока не правилось
> у тебя статуэтку можно отдавать Гораксу до того как Дарон отправить героя отнести статуэтку в Монастырь.
ты делаешь проверку по MIS_Addon_Daron_GetStatue == LOG_Running если добавить то такой ситуации не будет Npc_KnowsInfo(other,DIA_Addon_Daron_FoundStatue)

**Comment by dosinabox** [comment]
Хотя при MIS_OLDWORLD == LOG_SUCCESS ГГ просто отдаст статуэтку Дарону в диалоге DIA_Addon_Daron_FoundStatue. Гораксу следует оставить такую возможность только на случай смерти Дарона (?).
```cpp
func int DIA_Addon_Gorax_DaronsStatue_Condition()
{
	if((other.guild == GIL_NOV) && (DIA_Gorax_GOLD_perm == FALSE) && (Pedro_NOV_Aufnahme_LostInnosStatue_Daron == TRUE))
	{
		return TRUE;
	};
	if(Npc_IsDead(Daron) && (MIS_Addon_Daron_GetStatue == LOG_Running) && (LostInnosStatueInMonastery == FALSE))
	{
		return TRUE;
	};
};
```

**Comment by Kor-Angar** [comment]
вот если герой просто отдаст статуэтку Дарону это будет прекрасно. 

в случае смерти Дарона можно наверное и без проверки на MIS_Addon_Daron_GetStatue == LOG_Running

а у тебя нет проверки на саму статуэтки? :-)

**Comment by dosinabox** [comment]
> вот если герой просто отдаст статуэтку Дарону это будет прекрасно.

В целом, так и есть. Если ГГ маг или отдал письмо Гаронда Хагену, то Дарон просто забирает статуэтку. В остальных случаях посылает отнести её в монастырь. Только у Горакса надо условие подправить.

> в случае смерти Дарона можно наверное и без проверки на MIS_Addon_Daron_GetStatue == LOG_Running

Только от Дарона ГГ узнает о том, что статуэтка должна быть в монастыре. В этом же диалоге MIS_Addon_Daron_GetStatue  становится LOG_Running.

> а у тебя нет проверки на саму статуэтки? :-)

На наличие при сдаче Гораксу? Нет, это проверяется в самом диалоге и есть реплика Горакса на случай её отстутсвия.

**Comment by dosinabox** [comment]
Так)

ГГ узнает от Дарона, что статуэтку ждали в монастыре:
```cpp
AI_Output(self,other,"DIA_Addon_Daron_GuildHelp_10_03");	//С континента в монастырь была отправлена драгоценная статуэтка. Но до нас она так и не дошла.
AI_Output(other,self,"DIA_Addon_Daron_GuildHelp_15_04");	//Корабль был ограблен?
AI_Output(self,other,"DIA_Addon_Daron_GuildHelp_10_05");	//Нет. Он в целости и сохранности прибыл в Хоринис. Я встретил его и забрал статуэтку.
AI_Output(self,other,"DIA_Addon_Daron_GuildHelp_10_06");	//Но когда я возвращался в монастырь, на меня напали гоблины и отобрали ее у меня.
```
ГГ соглашается найти статуэтку (MIS_Addon_Daron_GetStatue становится LOG_Running):
```cpp
AI_Output(other,self,"DIA_Addon_Daron_GuildHelp_auftrag_15_00");	//Я услышал достаточно. Я найду статуэтку.
```
Далее ГГ находит статуэтку. Уже на этом этапе он может сам пойти в монастырь и отдать её Педро без диалога с Дароном:
```cpp
AI_Output(other,self,"DIA_Addon_Pedro_Statuette_15_00");	//У меня есть вот эта статуэтка. Думаю, она пропала из монастыря.
```
**Вопрос:** почему ГГ не может так же отдать статуэтку Гораксу после бегства Педро, не показывая её Дарону? Там такая же проверка (MIS_Addon_Daron_GetStatue == LOG_Running) и ГГ знает, что статуэтка должна быть в монастыре, о чем даже говорит:
```cpp
AI_Output(other,self,"DIA_Addon_Gorax_DaronsStatue_15_00");	//Вот статуэтка, которую должен был принести в монастырь Дарон.
```
> у тебя статуэтку можно отдавать Гораксу до того как Дарон отправить героя отнести статуэтку в Монастырь

**Comment by dosinabox** [comment]
Наблюдение:

За передачу статуэтки Гораксу ГГ получает 300 опыта (XP_Addon_ReturnedLostInnosStatue_Daron).
За передачу статуэтки Педро ГГ получает 100 опыта (XP_Addon_ReportLostInnosStatue2Daron).
За сообщение Дарону о передаче статуэтки в монастырь ГГ получает 100 опыта (XP_Addon_ReportLostInnosStatue2Daron).

Похоже на путаницу. За передачу Гораксу или Педро нужно давать 300 опыта (XP_Addon_ReturnedLostInnosStatue_Daron), а за сообщение Дарону - ещё 100 (XP_Addon_ReportLostInnosStatue2Daron).

**Comment by Kor-Angar** [comment]
квест про статуэтку даёт Дарон, что он там говорил и когда я честно не помню. вот запись в дневнике
![20210914171431_1](https://user-images.githubusercontent.com/49514764/133274302-146fb1c5-ef85-4cd2-818f-f8561a5e71ed.jpg)
у Дарона сейчас диалога нет, у него там там пусто. но есть диалог у Горакса. а статуэтки у меня нет. есть ли она вообще в мире тоже вопрос. 
Что там с этой статуэткой в итоге делать я честно не знаю. квест вроде простой, а топику уже 2 года :-)

**Comment by dosinabox** [comment]
> у Дарона сейчас диалога нет, у него там там пусто. но есть диалог у Горакса. а статуэтки у меня нет.

Объяснение очень простое. Только для Горакса записана реплика на случай, если у ГГ нет статуэтки:
```cpp
AI_Output(self,other,"DIA_Addon_Gorax_DaronsStatue_14_04");	//Статуэтку? Я не вижу никакой статуэтки. Принеси ее мне, и тогда я поверю твоему рассказу.
```
В случае с Педро и Дароном наличие статуэтки у ГГ проверяется до диалога. Если её нет, то диалог с ними не показывается. А с Гораксом показывается, он в разговоре проверяет. Вот и возникла путаница.

**Comment by Kor-Angar** [comment]
может быть лучше добавлять в мир гоблинов в правой пещере в диалоге Дарона?
